### PR TITLE
Feature: Implement Model Validations and Unit Specs

### PR DESCRIPTION
**Pull request by axcell-0**
Summary: This pull request addresses project requirements by introducing new validations and unit specifications.

Changes Made:

User Model Validations:

Ensures 'Name' is not blank.
Validates 'PostsCounter' as a non-negative integer.
Post Model Validations:

Ensures 'Title' is not blank.
Limits 'Title' to 250 characters.
Validates 'CommentsCounter' and 'LikesCounter' as non-negative integers.
Unit Specs:

Adds comprehensive unit tests for User and Post model methods and validations, including 'like,' 'post,' 'comment,' and 'user' methods.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
```
Comment (TedLivist) [inline spec/models/post_spec.rb:1]: - [ ] Your tests are failing. Kindly go through them again and fix the errors.

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @axcell-0,

Good job so far!
✔️ Written tests for models
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] You have not added validations for the models. Kindly add the validation using the requirements of the project.

  Post model | User model
  :---:|:---:
  image | image

- [ ] There was an issue setting up your project due to an indentation error in the `config/database.yml` file. As you can see from the first image, the database definition on line 53 is not properly indented to follow the alignment of the upper and lower lines. This raises an error. Kindly fix that.

  image

  image

_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @axcell-0👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

image

## To Highlight 🎉
- All tests are passed. ✔️
- Good job with validations. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] https://github.com/axcell-0/Blog-app/blob/5c69d773e46560998eb4a14ce5a761ff900bfbad/config/database.yml#L51-L55
  - As requested by the previous reviewer there is an issue setting up your project due to an indentation error in the `config/database.yml` file. The test database definition on line 53 is not properly indented to follow the alignment of the upper and lower lines. This raises an error. Please make sure to fix this. (To fix the issue just remove one white space before the `database: blog` text)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @axcell-0, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Fixed the issue related to intendation ✅ 

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**